macros/base/start_print.cfg

```diff
@@ -403,14 +403,39 @@ gcode:
     {% endif %}
 
 
-[gcode_macro _MODULE_CUSTOM1]
+[gcode_macro _MODULE_EXTRUDER_HEATING_TAP]
 gcode:
-    # ---- CUSTOM Macro section
-    # this section is reserved for personal customized start actions, which can be combined with all other start actions
-    # in order to use this, create a new macro in overrides.cfg
-    # [gcode_macro _MODULE_CUSTOM1]
-    # gcode:
-    #   ## Your custom code here
+    # Heat the nozzle to tap probing temperature ontop of the purge bucket (if available)
+    # or in the middle of the bed and wait 30s to let the nozzle dilate and stabilize
+    {% set EXTRUDER_TEMP = printer["gcode_macro _USER_VARIABLES"].tap_max_probing_temp|float %}
+    {% set INITIAL_TOOL = printer["gcode_macro START_PRINT"].initial_tool %}
+
+    {% set verbose = printer["gcode_macro _USER_VARIABLES"].verbose %}
+    {% set status_leds_enabled = printer["gcode_macro _USER_VARIABLES"].status_leds_enabled %}
+    {% set ercf_enabled = printer["gcode_macro _USER_VARIABLES"].ercf_enabled %}
+    {% set purge_and_brush_enabled = printer["gcode_macro _USER_VARIABLES"].purge_and_brush_enabled %}
+    {% set purgeclean_servo_enabled = printer["gcode_macro _USER_VARIABLES"].purgeclean_servo_enabled %}
+
+    {% set St = printer["gcode_macro _USER_VARIABLES"].travel_speed * 60 %}
+    {% set Px, Py, Pz = printer["gcode_macro _USER_VARIABLES"].purge_bucket_xyz|map('float') %}
+
+    {% set max_x = printer.toolhead.axis_maximum.x|float %}
+    {% set max_y = printer.toolhead.axis_maximum.y|float %}
+
+    {% if status_leds_enabled %}
+        STATUS_LEDS COLOR="HEATING"
+    {% endif %}
+
+    _CONDITIONAL_MOVE_TO_PURGE_BUCKET
+    {% if not purge_and_brush_enabled %}
+        G0 X{max_x|int / 2} Y{max_y|int / 3} Z50 F{St}
+    {% endif %}
+
+    M109 S{EXTRUDER_TEMP}
```

Comment:
Since the safe extruder temp is usually a temperature that doesn't allow material to flow properly, I would suggest to only heat to this temperature and do not care for the purge bucket position, etc... in this module.

Keep the status leds heating color, keep the M109 and basically that's it

Reply:
Updated macro M109 S{safe_extruder_temp}